src/UX/Identification/ProviderInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php namespace ILIAS\UX\Identification;
+
+/**
+ * Class ProviderInterface
+ *
+ * @author Fabian Schmid <fs@studer-raimann.ch>
+ */
+interface ProviderInterface {
+
+	/**
+	 * @param string $identifier
+	 *
+	 * @return IdentificationInterface
+	 */
+	public function internal(string $identifier): IdentificationInterface;
```

Comment:
On the interface I dislike this name even more. It's telling me not much about what it's task is.